state/batch.go

```diff
@@ -476,7 +482,7 @@ func (s *State) ProcessAndStoreClosedBatch(ctx context.Context, processingCtx Pr
 		StateRoot:     processedBatch.NewStateRoot,
 		LocalExitRoot: processedBatch.NewLocalExitRoot,
 		AccInputHash:  processedBatch.NewAccInputHash,
-		BatchL2Data:   encodedTxs,
+		BatchL2Data:   *processingCtx.BatchL2Data,
```

Comment:
```suggestion
		BatchL2Data:   *BatchL2Data,
```
If processingCtx.BatchL2Data is nil, then will panic